src/components/badge/index.tsx

```diff
@@ -1,109 +1,115 @@
 import _ from 'lodash';
-import PropTypes from 'prop-types';
-import React from 'react';
-import {StyleSheet, Text, ViewPropTypes} from 'react-native';
+import React, {PureComponent} from 'react';
+import {ImageSourcePropType, ImageStyle, StyleProp, StyleSheet, Text, TextStyle, TouchableOpacityProps, ViewStyle, ViewProps} from 'react-native';
 import {View as AnimatableView} from 'react-native-animatable';
-import {PureBaseComponent} from '../../commons';
+import {extractAccessibilityProps, extractAnimationProps} from '../../commons/modifiers';
+import {asBaseComponent} from '../../commons/new';
 import {BorderRadiuses, Colors, ThemeManager, Typography} from '../../style';
 import Image from '../image';
 import TouchableOpacity from '../touchableOpacity';
 import View from '../view';
 
-const LABEL_FORMATTER_VALUES = [1, 2, 3, 4];
+const LABEL_FORMATTER_VALUES = [1, 2, 3, 4] as const;
 
 // TODO: depreciate enum badge sizes, use only number for size
-export const BADGE_SIZES = {
-  pimpleSmall: 6,
-  pimpleBig: 10,
-  pimpleHuge: 14,
-  small: 16,
-  default: 20,
-  large: 24
+export enum BADGE_SIZES {
+  pimpleSmall = 6,
+  pimpleBig = 10,
+  pimpleHuge = 14,
+  small = 16,
+  default = 20,
+  large = 24
 };
 
+type LabelFormatterValues = typeof LABEL_FORMATTER_VALUES[number]
+export type BadgeSizes = keyof typeof BADGE_SIZES
+
+export type BadgePropTypes = ViewProps & TouchableOpacityProps & {
```

Comment:
Let's keep the name as `BadgeProp` instead of `BadgePropTypes`, I wanted to have a shorter name 
(I know we already have some components with xxxPropTypes, we'll change them eventually).